runtime/doc/options.txt

```diff
@@ -4364,6 +4364,50 @@ A jump table for the options with a short description can be found at |Q_op|.
 	    |nobody         |  3 nobody     |  0 nobody     |3   nobody
 	    |there          |  4 there      |  1 there      |  1 there
 
+				*'numbercolumn'* *'nuc'*
+'numbercolumn' 'nuc'	string	(default: empty)
+			local to window
+	EXPERIMENTAL
+	When this option is not empty, it determines the content of the number
+	column. The format of this option is like that of 'statusline'. The
+	variables *v:redraw_lnum* and *v:redraw_relnum* will be set to the to be
+	printed line number and relative line number respectively. These can be
+	evaluated in an expression inside a 'statusline' %{} block.
+
+	Some of the items from the 'statusline' format are different
+	'numbercolumn':
+
+	%l	v:redraw_lnum, can be used inside %{} block
```

Comment:
This is wrong `%l` cannot be used in `%{}`. Make sure to document `%l` and `v:redraw_lnum` clearly.

Reply:
It's not wrong in this PR. I have altered the `%{}` eval block to substitutie `v:redraw_lnum` and `v:redraw_relnum` for `%l` and `%r` respectively: https://github.com/luukvbaal/neovim/blob/93a9277b76b76e0a0028ebf5aa5136b6418ce1e7/src/nvim/statusline.c#L1152-L1158 (see https://github.com/neovim/neovim/pull/20621#discussion_r998503306)

This allows you to write e.g. `set nuc=%{%r?%r:%l}` instead of `set nuc=%{v:redraw_relnum?v:redraw_relnum:v:redraw_lnum}`. Might not be worth it.

Reply:
Not a fan personally. I'd reserve `%{}` blocks for viml expressions.

Reply:
> Not a fan personally. I'd reserve %{} blocks for viml expressions.

Agreed. We shouldn't change the behavior of `%{}`.

> This allows you to write e.g. `set nuc=%{%r?%r:%l}` instead of `set nuc=%{v:redraw_relnum?v:redraw_relnum:v:redraw_lnum}`

There's already a syntax for re-evaluating the expression result - `%{%%}`. Wouldn't something like `%{%'%r?%r:%l'%}` acheive the same result? 🤔

Reply:
No that wouldn't work. The number needs to be substituted before the viml expression is evaluated. But this change has felt dubious to me from the start as well. I'll revert it.